lib/driver.js

```diff
@@ -382,7 +380,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startWda (sessionId, realDevice) {
-    this.wda = new WebDriverAgent(this.xcodeVersion, this.opts);
+    this.wda = new WebDriverAgent(this.xcodeVersion, Object.assign(this.opts, {isFirstSession: this.isFirstSession}));
```

Comment:
I think what @imurchie is saying is that your code is equivalent to:

```js
this.opts.isFirstSession = this.isFirstSession;
this.wda = new WebDriverAgent(this.xcodeVersion, this.opts);
```
which means we do give WDA the opportunity to mutate `this.opts` if it so desires.